yt_dlp/compat/urllib/request.py

```diff
@@ -0,0 +1,36 @@
+# flake8: noqa: F405
+from urllib.request import *  # noqa: F403
+from .. import compat_os_name
+from ..compat_utils import passthrough_module
+import sys
+
+passthrough_module(__name__, 'urllib.request')
+del passthrough_module
+
+
+if compat_os_name == 'nt':
+    # On older python versions, proxies are extracted from Windows registry erroneously. [1]
+    # If the https proxy in the registry does not have a scheme, urllib will incorrectly add https:// to it. [2]
+    # It is unlikely that the user has actually set it to be https, so we should be fine to safely downgrade
+    # it to http on these older python versions to avoid issues
+    # This also applies for ftp proxy type, as ftp:// proxy scheme is not supported.
+    # 1: https://github.com/python/cpython/issues/86793
+    # 2: https://github.com/python/cpython/blob/51f1ae5ceb0673316c4e4b0175384e892e33cc6e/Lib/urllib/request.py#L2683-L2698
+    from urllib.request import getproxies_environment, getproxies_registry
+
+    def getproxies_registry_patched():
+        proxies = getproxies_registry()
+        if (
+            sys.version_info >= (3, 10, 5)  # https://docs.python.org/3.10/whatsnew/changelog.html#python-3-10-5-final
+            or (3, 9, 13) <= sys.version_info < (3, 10)  # https://docs.python.org/3.9/whatsnew/changelog.html#python-3-9-13-final
```

Comment:
I think it makes sense to start with lowest version and always use `<=`/`<`
```suggestion
            (3, 9, 13) <= sys.version_info < (3, 10)  # https://docs.python.org/3.9/whatsnew/changelog.html#python-3-9-13-final
            or (3, 10, 5) <= sys.version_info  # https://docs.python.org/3.10/whatsnew/changelog.html#python-3-10-5-final
```
Maybe it could also hint at `bpo-42627` to make it easier to `<CTRL>+F`?

Reply:
don't think it really matters -  I kinda prefer the current way :upside_down_face: 

I have already linked the python issue which should be enough